cuda_core/tests/memory_ipc/test_errors.py

```diff
@@ -0,0 +1,139 @@
+        except Exception as e:
+            exc_info = type(e), str(e)
+        else:
+            exc_info = None, None
+        pipe[1].put(exc_info)
+
+
+class TestAllocFromImportedMr(ChildErrorHarness):
+    """Error when attempting to allocate from an import memory resource."""
+
+    def PARENT_ACTION(self, queue):
+        queue.put(self.mr)
+
+    def CHILD_ACTION(self, queue):
+        mr = queue.get(timeout=CHILD_TIMEOUT_SEC)
+        mr.allocate(NBYTES)
+
+    def ASSERT(self, exc_type, exc_msg):
+        assert exc_type is TypeError
+        assert exc_msg == "Cannot allocate from shared memory pool imported via IPC"
+
+
+class TestImportWrongMR(ChildErrorHarness):
+    """Error when importing a buffer from the wrong memory resource."""
+
+    def PARENT_ACTION(self, queue):
+        options = DeviceMemoryResourceOptions(max_size=POOL_SIZE, ipc_enabled=True)
+        mr2 = DeviceMemoryResource(self.device, options=options)
+        buffer = mr2.allocate(NBYTES)
+        queue.put([self.mr, buffer.get_ipc_descriptor()])  # Note: mr does not own this buffer
+
+    def CHILD_ACTION(self, queue):
+        mr, buffer_desc = queue.get(timeout=CHILD_TIMEOUT_SEC)
+        Buffer.from_ipc_descriptor(mr, buffer_desc)
+
+    def ASSERT(self, exc_type, exc_msg):
+        assert exc_type is CUDAError
+        assert "CUDA_ERROR_INVALID_VALUE" in exc_msg
+
+
+class TestExportImportedMR(ChildErrorHarness):
+    """Error when exporting a memory resource that was imported."""
+
+    def PARENT_ACTION(self, queue):
+        queue.put(self.mr)
+
+    def CHILD_ACTION(self, queue):
+        mr = queue.get(timeout=CHILD_TIMEOUT_SEC)
+        mr.get_allocation_handle()
+
+    def ASSERT(self, exc_type, exc_msg):
+        assert exc_type is RuntimeError
+        assert exc_msg == "Imported memory resource cannot be exported"
+
+
+class TestImportBuffer(ChildErrorHarness):
+    """Error when using a buffer as a buffer descriptor."""
+
+    def PARENT_ACTION(self, queue):
+        # Note: if the buffer is not attached to something to prolong its life,
+        # CUDA_ERROR_INVALID_CONTEXT is raised from Buffer.__del__
+        self.buffer = self.mr.allocate(NBYTES)
+        queue.put(self.buffer)
+
+    def CHILD_ACTION(self, queue):
+        buffer = queue.get(timeout=CHILD_TIMEOUT_SEC)
+        Buffer.from_ipc_descriptor(self.mr, buffer)
+
+    def ASSERT(self, exc_type, exc_msg):
+        assert exc_type is TypeError
+        assert exc_msg.startswith("Argument 'ipc_buffer' has incorrect type")
+
+
+class TestDanglingBuffer(ChildErrorHarness):
+    """
+    Error when importing a buffer object without registering its memory
+    resource.
+    """
+
+    def PARENT_ACTION(self, queue):
+        options = DeviceMemoryResourceOptions(max_size=POOL_SIZE, ipc_enabled=True)
+        mr2 = DeviceMemoryResource(self.device, options=options)
+        self.buffer = mr2.allocate(NBYTES)
+        queue.put(self.buffer)  # Note: mr2 not sent
```

Comment:
I am a bit confused by this test: I thought we allow sending a single buffer over, and the register/serialization mechanism would ensure the MR is accessible on the child side? Maybe I get confused with another use case?